openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -416,7 +417,10 @@ def post(self, request):
 
                 # Log the user out.
                 logout(request)
-            return Response(status=status.HTTP_204_NO_CONTENT)
+            response = Response(status=status.HTTP_204_NO_CONTENT)
+            if not request.user.is_authenticated:
```

Comment:
@burhandodhy why we need this `if` statement?

Reply:
I misunderstand the `transaction.atomic()` working. I thought it returns to function if any query failed and to make sure that the user is logout I added the `if` condition. 

Going to remove the if condition.

Reply:
@ziafazal `if` is removed.